apps/projects/app/utils/apollo-client.js

```diff
@@ -1,16 +1,34 @@
 import ApolloClient from 'apollo-boost'
+import { useAragonApi } from '../api-react'
+import { GITHUB_TOKEN_REVOKED } from '../store/eventTypes'
+import { STATUS } from '../utils/github'
 
-export const initApolloClient = token =>
-  new ApolloClient({
-    uri: 'https://api.github.com/graphql',
-    request: operation => {
-      if (token) {
-        operation.setContext({
-          headers: {
-            accept: 'application/vnd.github.starfire-preview+json', // needed to create issues
-            authorization: `bearer ${token}`,
-          },
-        })
+export const useApolloClient = () => {
+  const { api } = useAragonApi()
+  const initApolloClient = token => {
+    if (token === null) return null
+    return new ApolloClient({
+      uri: 'https://api.github.com/graphql',
+      request: operation => {
+        if (token) {
+          operation.setContext({
+            headers: {
+              accept: 'application/vnd.github.starfire-preview+json', // needed to create issues
+              authorization: `bearer ${token}`,
+            },
+          })
+        }
+      },
+      onError: error => {
+        if (error.networkError && error.networkError.statusCode === 401) {
+          api.emitTrigger(GITHUB_TOKEN_REVOKED, {
+            status: STATUS.REVOKED,
+            scope: null,
+            token: null,
+          })
+        }
       }
-    },
-  })
+    })
+  }
+  return initApolloClient
```

Comment:
I think we can save all consumers of `useApolloClient` a line or two by doing something like this:

```js
export const useApolloClient = () => {
  const { api, appState } = useAragonApi()
  if (appState.github.token === null) return null
  return new ApolloClient({ ... })
}
```

Note that there's also no reason to pass `token` in as an argument anymore, now that this is a hook!

Do you think this would work? Is there a reason to make this return an `initApolloClient` function, rather than `null` or an actual client?